flow360/plugins/report/report_items.py

```diff
@@ -1211,6 +1229,34 @@ def _get_figures(self, cases, context: ReportContext):
 
         return file_names, data.x_label, data.y_label
 
+    def _check_2d_caption_validity(self, cases):
+        """Checks the validity of caption input for a given Chart2D instance."""
+
+        if isinstance(self.caption, PatternCaption):
+            raise ValueError("PatternCaption is not supported for Chart2D.")
+
+        if isinstance(self.caption, List):
+            if self.separate_plots is not True:
+                raise ValueError(
+                    "List of captions is only supported for Chart2D when separate_plots is True."
+                )
+            if len(self.caption) != len(cases):
+                raise ValueError("Caption list is not the same length as the list of cases.")
+
+    def _handle_2d_caption(self, **kwargs):
+        """Handle captions for Chart2D."""
+
+        case_number = kwargs.get("case_number", None)
+
+        if isinstance(self.caption, List):
+            if isinstance(case_number, int):
+                return self.caption[case_number]
+            raise ValueError(
+                "For list of captions, case number of type integer needs to be provided."
+            )
```

Comment:
although this is correct, I think it is redundant and it is information for developer not user, thus I would change to:
```suggestion
            return self.caption[case_number]
```